### PR TITLE
feat(colors): synchronise with figma

### PR DESCRIPTION
**Pull request by matthprost**
## Type

- Feature

#### What is expected?

To sync figma tokens with our project. We yet do not expose dark theme as all components still need to be reviewed with new colors.

⚠️ This MR only contains synchronised files, old shades match and new synchronised colors are living together. In another MR we will review components one by one in order to end synchronisation with design team. 

#### The following changes where made:

1. Added ci for synchronise design tokens from figma and create PR automatically

2. Added theme wrapper into theme documentation to make colors change depending on theme

### CODE REVIEW

File: .github/workflows/sync-design-tokens.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+          URL: https://raw.githubusercontent.com/scaleway/design-tokens/main/tokens.json
+        run: |
+          curl ${URL} | jq 'reduce (.dark.shades | to_entries | .[]) as $sentiment ({}; . + {"\($sentiment.key)": (reduce($sentiment.value | to_entries | .[]) as $shade ({}; . + { "\($shade.key)": $shade.value.value })) }) | {"colors": .}' > src/theme/tokens/dark.json
+          curl ${URL} | jq 'reduce (.light.shades | to_entries | .[]) as $sentiment ({}; . + {"\($sentiment.key)": (reduce($sentiment.value | to_entries | .[]) as $shade ({}; . + { "\($shade.key)": $shade.value.value })) }) | {"colors": .}' > src/theme/tokens/light.json
+
+      - name: Compare & Create Pull Request
+        uses: peter-evans/create-pull-request@v3
+        with:
+          base: main
+          author: design <design@users.noreply.github.com>
```
Comment (chambo-e) [inline .github/workflows/sync-design-tokens.yml:25]:
```suggestion
          author: Scaleway Design Bot <github@scaleway.com>
```

File: .github/workflows/sync-design-tokens.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+          URL: https://raw.githubusercontent.com/scaleway/design-tokens/main/tokens.json
+        run: |
+          curl ${URL} | jq 'reduce (.dark.shades | to_entries | .[]) as $sentiment ({}; . + {"\($sentiment.key)": (reduce($sentiment.value | to_entries | .[]) as $shade ({}; . + { "\($shade.key)": $shade.value.value })) }) | {"colors": .}' > src/theme/tokens/dark.json
+          curl ${URL} | jq 'reduce (.light.shades | to_entries | .[]) as $sentiment ({}; . + {"\($sentiment.key)": (reduce($sentiment.value | to_entries | .[]) as $shade ({}; . + { "\($shade.key)": $shade.value.value })) }) | {"colors": .}' > src/theme/tokens/light.json
+
+      - name: Compare & Create Pull Request
+        uses: peter-evans/create-pull-request@v3
+        with:
+          base: main
+          author: Scaleway Design Bot <design@scaleway.com>
```
Comment (chambo-e) [inline .github/workflows/sync-design-tokens.yml:25]: Needs to be github@scaleway.com to be associated with https://github.com/scaleway-bot :)

File: .github/workflows/sync-design-tokens.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+          commit-message: 'style(theme): synchronise design tokens'
```
Comment (chambo-e) [inline .github/workflows/sync-design-tokens.yml:26]: style is for code style, not really design, we'll also need to deploy a new version each time, so a feat or fix should one used :)

File: .github/workflows/sync-design-tokens.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+          branch: style/design-tokens-synchronise
+          delete-branch: true
+          title: 'style(theme): synchronise design tokens'
```
Comment (chambo-e) [inline .github/workflows/sync-design-tokens.yml:29]: Title will be based on message commit no ?

File: .storybook/preview.tsx [hunk 5]
```diff
@@ -73,22 +99,33 @@ export const globalStyles = (theme: Theme) => css`
...
+        defaultTranslations={{}}
+        enableDebugKey={false}
+        enableDefaultLocale={false}
+        loadDateLocale={async locale =>
+          import(`date-fns/locale/${locale}/index`)
+        }
+        localeItemStorage="localeI18n"
+        supportedLocales={['en', 'fr', 'es']}
+      >
+        <ThemeProvider theme={generatedTheme}>
```
Comment (chambo-e) [inline .storybook/preview.tsx:124]: Can we fix this error ?

File: package.json [hunk 6]
```diff
@@ -106,6 +106,7 @@
     "@emotion/react": "11.7.1",
     "@emotion/styled": "11.6.0",
     "@rollup/plugin-babel": "5.3.0",
+    "@rollup/plugin-json": "^4.1.0",
```
Comment (chambo-e) [inline package.json:109]:
```suggestion
    "@rollup/plugin-json": "4.1.0",
```

File: package.json [hunk 7]
```diff
@@ -165,6 +166,7 @@
     "semantic-release": "18.0.1",
     "shx": "0.3.4",
     "size-limit": "7.0.8",
+    "storybook-dark-mode": "^1.0.8",
```
Comment (chambo-e) [inline package.json:169]:
```suggestion
    "storybook-dark-mode": "1.0.8",
```

File: package.json [hunk 8]
```diff
@@ -123,7 +124,7 @@
     "@storybook/addon-actions": "6.4.19",
     "@storybook/addon-essentials": "6.4.19",
     "@storybook/addon-links": "6.4.19",
-    "@storybook/addons": "6.4.19",
+    "@storybook/addons": "^6.4.19",
```
Comment (chambo-e) [inline package.json:127]:
```suggestion
    "@storybook/addons": "6.4.19",
```

File: rollup.config.mjs [hunk 9]
```diff
@@ -72,6 +73,7 @@ export default [{
         gzipSize: true,
         open: true,
       }),
+    json()
```
Comment (chambo-e) [inline rollup.config.mjs:76]: Could we generate ts files instead of json to avoid the added complexity ?

File: src/theme/colors/index.ts [hunk 10]
```diff
@@ -1,4 +1,9 @@
-import * as localContrasts from './contrasts'
+import * as light from '../tokens/light.json'
+
+// Scaleway UI is based on light colors by default
+const localContrasts = (
+  light as Record<string, Record<string, Partial<ContrastType>>>
```
Comment (chambo-e) [inline src/theme/colors/index.ts:5]: ContrastType is declared after

**Review by chambo-e — CHANGES_REQUESTED** [review]

File: .github/workflows/sync-design-tokens.yml (hunk 4 above)
Comment (matthprost) [reply .github/workflows/sync-design-tokens.yml:29]: If not title written github action is putting its default value so I have to add one

File: .github/workflows/sync-design-tokens.yml (hunk 3 above)
Comment (matthprost) [reply .github/workflows/sync-design-tokens.yml:26]: Alright I wasn't sure about it thanks for clarifying :)

File: src/index.ts [hunk 11]
```diff
@@ -2,4 +2,4 @@ export * from './components'
 export { default as theme } from './theme'
 export type { SCWUITheme } from './theme'
 export * from './utils'
-export { createTheme } from './theme'
+export { createTheme, darkTheme } from './theme'
```
Comment (chambo-e) [inline src/index.ts:5]:
```suggestion
export { default as lightTheme, createTheme, darkTheme } from './theme'
```

File: rollup.config.mjs (hunk 9 above)
Comment (matthprost) [reply rollup.config.mjs]: @chambo-e Sure!

File: scripts/figma-synchronise-tokens.sh [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+   (printf "/**
+ * Provides all tokens of a specific theme edited by design team.
+ * This file is automatically generated from /scripts/figma-synchronise-tokens.sh.
+ * PLEASE DO NOT EDIT HERE
+ */
+ /* eslint-disable sort-keys */
+export default "; echo "${FINAL_RESULT}"; echo "/* eslint-enable sort-keys */") > src/theme/tokens/"${i}".ts
+done
+
+yarn run format
```
Comment (chambo-e) [inline scripts/figma-synchronise-tokens.sh:85]: Can we run this command outside of the script ?

File: src/index.ts [hunk 13]
```diff
@@ -1,5 +1,4 @@
 export * from './components'
-export { default as theme } from './theme'
+export { default as theme, extendTheme, createTheme } from './theme'
```
Comment (chambo-e) [inline src/index.ts:2]: Do we still use createTheme ?

**Review by chambo-e — CHANGES_REQUESTED** [review]

File: tsconfig.json [hunk 14]
```diff
@@ -9,7 +9,8 @@
     "noEmit": true,
     "moduleResolution": "node",
     "skipLibCheck": true,
-    "jsx": "preserve"
+    "jsx": "preserve",
+    "resolveJsonModule": true
```
Comment (chambo-e) [inline tsconfig.json:13]: Still needed ?

File: scripts/figma-synchronise-tokens.sh (hunk 12 above)
Comment (matthprost) [reply scripts/figma-synchronise-tokens.sh:85]: right good one

File: tsconfig.json (hunk 14 above)
Comment (matthprost) [reply tsconfig.json:13]: Not needed you're right

File: .storybook/preview.tsx [hunk 15]
```diff
@@ -1,20 +1,30 @@
...
 if (process.env.STORYBOOK_ENVIRONMENT === 'visual') mockRandom([0.25, 0.5])
 
 const STORY_SORT = {
   order: ['Home', 'Testing', 'Theme', 'Components'],
 }
 
+const darkMode = {
+  dark: { ...themes.dark, ...dark },
+  light: { ...themes.normal, ...light },
+}
```
Comment (QuiiBz) [inline .storybook/preview.tsx:23]: Strange to have a variable named `darkMode` with both dark and light themes 🤔

File: tsconfig.json [hunk 16]
```diff
@@ -9,7 +9,7 @@
     "noEmit": true,
     "moduleResolution": "node",
     "skipLibCheck": true,
-    "jsx": "preserve"
+    "jsx": "preserve",
```
Comment (QuiiBz) [inline tsconfig.json:12]: JSON shouldn't have a trailing comma: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Trailing_commas#trailing_commas_in_json

File: .storybook/preview.tsx (hunk 15 above)
Comment (matthprost) [reply .storybook/preview.tsx:23]: @QuiiBz Don't ask me thats the variable name of storybook haha

File: src/components/UnitInput/index.tsx [hunk 17]
```diff
@@ -31,7 +32,8 @@ const CustomRichSelect = styled(RichSelect)`
   &:hover,
   &:focus {
     text-decoration: none;
-    border-color: ${({ theme }) => theme.colors.primary.borderWeak};
+    border-color: ${({ theme }: { theme: SCWUITheme }) =>
```
Comment (chambo-e) [inline src/components/UnitInput/index.tsx:35]: Why ? 🤔

**Review by DorianMaliszewski — APPROVED** [review]

**Review by QuiiBz — APPROVED** [review]

File: src/components/UnitInput/index.tsx (hunk 17 above)
Comment (matthprost) [reply src/components/UnitInput/index.tsx:35]: It was a test I forgot to remove

**Pull request merged**